src/main/java/com/manardenza/service/ReservationService.java

```diff
@@ -0,0 +1,60 @@
+package com.manardenza.service;
+
+import com.manardenza.dao.ReservationDaoImpl;
+import com.manardenza.entity.Reservation;
+import com.manardenza.entity.Room;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public final class ReservationService {
+
+    private static ReservationService instance;
+    private ReservationDaoImpl reservationDao;
+
+    ReservationService() {
+        this.reservationDao = ReservationDaoImpl.getInstance();
+    }
+
+    public static ReservationService getInstance() {
+        if (instance == null) {
+            instance = new ReservationService();
+        }
+        return instance;
+    }
+
+    public Long bookRoom(Reservation reservation) {
+        return reservationDao.save(reservation).getId();
+    }
+
+    public boolean cancelReservation(long idReservation) {
+        return reservationDao.getAll().removeIf(p -> p.getId() == idReservation);
+    }
+
+    public Map<String, List<Room>> checkRoomReservation(Date reservedFrom, Date reservedTo,
+                                                        Map<String, List<Room>> rooms) {
+        for (String key : rooms.keySet()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2864)